tests/math_builtin_api/modules/test_generator.py

```diff
@@ -153,22 +154,26 @@ def generate_test_cases(test_id, types, sig_list):
 # We attempt to find a combination that is part of the current generic type (e.g. floatn)
 
 def attempt_match(runner, var_type, base_type, dim, unsigned, current_type):
+    ct_keys = list(current_type.keys())
+    ct_part_keys = []
+    for t in ct_keys:
```

Comment:
Would this work?
```suggestion
    for t in current_type.keys():
```

Reply:
Yes, it works, thank you! Removed list in other places too.